bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/SecretsVaultService.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link SecretsVaultService} can be used to store and retrieve secrets.
+ *
+ * Secrets can only be retrieved by the bundle that stored them and by default
```

Comment:
> Secrets can only be retrieved by the **bundle that stored them**

How this is handled? Should secret storing and modifying be possible e.g. from karaf console (not bundle which will use the secrets)?

Reply:
This is possible via the management interface of this proposal, that is also used by the rest interface. Retrieving is not possible. Modifying requires to supply the old secret.